Allow the component and action to also be set by the library

sixlive opened the pull request:
## Description
By allowing the component and action to be set by the context as well as the additional parameters we can leverage it to easily set the component and action for the Laravel library.

```php
$badger = Honeybadger::new([
    'api_key' => 'asdf',
    'handlers' => [
        'exception' => false,
        'error' => false,
    ],
], $client->make());

$badger->setComponent('HomeController');
$badger->setAction('index');
```

## Todos
- [x] Tests
- [ ] Documentation
- [ ] Changelog Entry (unreleased)

## Steps to Test or Reproduce
Outline the steps to test or reproduce the PR here.

```bash
> git pull --prune
> git checkout <branch>
> vendor/bin/phpunit
```

sixlive commented:
Maybe we should prefix them with honeybadger_ ?

joshuap commented:
Yeah, I could go for that. Do you think we should delete the from the context when reading them, so they don't appear in the context section?

sixlive commented:
@joshuap I made it much more simple. I added a `setComponent` and `setAction` method to the Honeybadger object. This still uses context under the hood but once we were talking about prefixing I didn't want to have to expose that implementation detail to a consumer. I also updated it so that we strip these values from the context since it really doesn't make sense to be in 2 places.
